enhancements/storage/csi-driver-install.md

```diff
@@ -0,0 +1,181 @@
+  * Right now it's implementation detail, this enhancement focuses on CSI driver installation. Proper enhancement will be created for CSI migration.
+
+## Proposal
+
+We are currently considering using OLM for installation of CSI driver.
+
+### Installation via OLM
+
+We propose that - we provide each driver mentioned above as a separate operator which could be subscribed and installed via OLM UI. Each driver's operator
+is responsible for its installation and release. The operator is responsible for creating storageclass that the driver provides.
+
+The configuration of CSI driver can be done via OLM UI if required and CSI driver can access cloudprovider credentials from Openshift provided sources.The CR that is responsible for driver configuration can be installed by default by the operator itself.
+
+Installation via OLM however means that, when we want to enable these CSI drivers as default drivers, they must be installed by default in Openshift installs.
+We further propose that - Cluster Storage Operator(https://github.com/openshift/cluster-storage-operator) could create subscriptions for these driver operators when drivers have to be installed by default.
+
+Expected workflow as optional driver (using EBS as an example):
+1. User finds EBS CSI driver in OLM and installs it.
+2. EBS CSI driver is installed and it creates relevant storageclass that user can use to provision CSI EBS volumes.
+
+Expected workflow as default driver:
+1. CVO installs cluster-storage-operator.
+2. cluster-storage-operator detects cloudprovider on which cluster is running(lets say EBS).
+3. cluster-storage-operator creates a subscription for EBS CSI driver using redhat operator source.
+4. cluster-storage-operator monitors progress of subscription and sets its own status as available when subscription is installed.
+
+cluster-storage-operator ensures that there is always an subscription for CSI driver in given cloudprovider environment.
+
+
+There are pros and cons to this approach and since this KEP is in its first iteration, we would like to seek feedback from other stakeholders.
+
+Pros:
+1. It is simple to create a opt-in installable driver operator.
+2. Providing an optional UI for configuration via OLM is a plus.
+3. Each driver's operator sits in its own repository and managing releases could be simpler.
+
+Cons:
+1. When these CSI drivers have to be installed by default, this approach causes a coupling between CVO managed cluster-storage-operator and driver's operator.
+
+When a CSI driver operator is in technical preview, we expect that the operator will be available from a `beta` channel. Moving to a `stable` channel once a driver reaches GA will require Openshift admin to manually change subscribed channel from beta to stable. At this point we expect that, operator in GA state will simply adopt the resources(CRs) created by beta version of the operator.
```

Comment:
So are we going to have the beta channel open forever? Because I can image users won't switch the channel manually. And we should keep the cluster running. It is a viable option, however, it should be noted in the proposal.

Reply:
Once CVO adopts an operator's subscription then it should update the operator's channel to whatever cluster-storage-operator is configured to.

Reply:
Good idea!